src/Library/demos/Stack/main.vala

```diff
@@ -0,0 +1,60 @@
+#! /usr/bin/env -S vala workbench.vala --pkg gtk4 --pkg libadwaita-1 --pkg gio-2.0
+
+using Gtk;
+using Adw;
+using Gdk;
+using GLib;
+
+public void main() {
+  var root_box = workbench.builder.get_object("root_box") as Box;
+  var stack = workbench.builder.get_object("stack") as Stack;
+  var navigation_row = workbench.builder.get_object("navigation_row") as ComboRow;
+  var transition_row = workbench.builder.get_object("transition_row") as ComboRow;
+
+  Gtk.Separator? separator = null;
+  Gtk.StackSwitcher? stack_switcher = null;
+  Gtk.StackSidebar? stack_sidebar = null;
+
+  if (navigation_row.selected == 0) {
+    stack_switcher = new Gtk.StackSwitcher();
+    stack_switcher.stack = stack;
+    root_box.prepend(stack_switcher);
+  } else {
+    stack_sidebar = new Gtk.StackSidebar();
+    separator = new Separator(Gtk.Orientation.HORIZONTAL);
+    stack_sidebar.stack = stack;
+    root_box.prepend(separator);
+    root_box.prepend(stack_sidebar);
+  }
+
+  navigation_row.notify["selected"].connect(() => {
+    if (navigation_row.selected == 0) {
```

Comment:
Can you check if the string in the item is correct, instead of just the position. That would be better readable.